sql/parser/walk.go

```diff
@@ -162,8 +162,34 @@ func (v *argVisitor) Visit(expr Expr) Expr {
 
 // FillArgs replaces any placeholder nodes in the expression with arguments
 // supplied with the query.
-func FillArgs(expr Expr, args Args) (Expr, error) {
+func FillArgs(stmt Statement, args Args) error {
 	v := argVisitor{args: args}
-	expr = WalkExpr(&v, expr)
-	return expr, v.err
+	WalkStmt(&v, stmt)
+	return v.err
+}
+
+// WalkStmt walks the entire parsed stmt calling WalkExpr on each
+// expression, and replacing each expression with the one returned
+// by WalkExpr.
+func WalkStmt(v Visitor, stmt Statement) {
+	switch stmt := stmt.(type) {
+	case *Select:
+		for _, expr := range stmt.Exprs {
+			switch expr := expr.(type) {
+			case *NonStarExpr:
+				expr.Expr = WalkExpr(v, expr.Expr)
+			}
+		}
+		if stmt.Where != nil {
+			stmt.Where.Expr = WalkExpr(v, stmt.Where.Expr)
+		}
```

Comment:
Add a `TODO` (or implement) the remaining portions of `parser.Select`. In particular, you're missing `stmt.Having`, `stmt.GroupBy`, `stmt.OrderBy` and `stmt.Limit`.